lib/commands/deploy.ts

```diff
@@ -44,16 +43,17 @@ export class DeployHelper implements IDeployHelper {
 
 			let action = (device: Mobile.IDevice): IFuture<void> => {
 				let deploymentTarget = this.$project.projectData.iOSDeploymentTarget;
-				if(deploymentTarget && device.deviceInfo.platform.toLowerCase() === this.$devicePlatformsConstants.iOS.toLowerCase()) {
+				if (deploymentTarget && device.deviceInfo.platform.toLowerCase() === this.$devicePlatformsConstants.iOS.toLowerCase()) {
 					let deviceVersion = _.take(device.deviceInfo.version.split("."), 2).join(".");
-					if(helpers.versionCompare(deviceVersion, deploymentTarget) < 0) {
+					if (helpers.versionCompare(deviceVersion, deploymentTarget) < 0) {
 						this.$logger.error(`You cannot deploy on device ${device.deviceInfo.identifier} with OS version ${deviceVersion} when iOSDeploymentTarget is set to ${deploymentTarget}.`);
 						return Future.fromResult();
 					}
 				}
 
-				if(!packageFile) {
+				if (!packageFile) {
 					let packageDefs = this.$buildService.deploy(this.$devicesServices.platform, device).wait();
+					packageDefs = _.filter(packageDefs, (def: Server.IPackageDef) => !def.disposition || def.disposition === "BuildResult");
```

Comment:
I really could not name this method nor did I found its proper place :( `getBinaryPackage`?